### PR TITLE
Cleanup internal events and tasks that are delegated to Hyades

### PR DESCRIPTION
**Pull request by nscuro**
### Description

This PR removes internal `Event`s and `Task`s related to repository meta and vulnerability analysis, as these two domains are now externalized to Hyades services.

The PR includes #14, because touches a few areas that I also had to modify in that PR. Would make sense to merge #14 first.

For both repository meta and vulnerability analysis, we still have two `Event`s that we process internally:

* `PortfolioRepositoryMetaAnalysisEvent`
* `ProjectRepositoryMetaAnalysisEvent`
* `PortfolioVulnerabilityAnalysisEvent`
* `ProjectVulnerabilityAnalysisEvent`

The reason is that fetching all components for large projects or the entire portfolio, and dispatching Kafka events for every fetched component, can be a costly operation. It is still beneficial to do this asynchronously. In cases where we need to dispatch only a few events for a single component, we immediately dispatch to Kafka, without publishing an internal event first.

### Additional Details

`fetchNextComponentsPage` used in both `ReposioryMetaAnalysisTask` and `VulnerabilityAnalysisTask` is duplicated intentionally. It probably makes sense to move this logic to a shared class, but I didn't want to generalize too much for now.

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [ ] ~This PR fixes a defect, and I have provided tests to verify that the fix is effective~
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] ~This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)~
- [ ] ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

File: src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java [hunk 1]
```diff
@@ -21,176 +21,119 @@
...
+                filter += " && project == :project";
+                params.put("project", project);
+            }
             if (lastId != null) {
-                query.setFilter("id < :lastId");
-                query.setParameters(lastId);
+                filter += " && id < :lastId";
+                params.put("lastId", lastId);
             }
+            query.setFilter(filter);
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java:135]: Filter is built without dynamic variables and uses parameters.

[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java:135]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            params.put("projectActive", true);
+            if (project != null) {
+                filter += " && project == :project";
+                params.put("project", project);
+            }
+            if (lastId != null) {
+                filter += " && id < :lastId";
+                params.put("lastId", lastId);
+            }
+            query.setFilter(filter);
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java:132]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            } catch (Exception ex) {
+                LOGGER.error("An unexpected error occurred while submitting components for repository meta analysis", ex);
+            }
+        }
+    }
+
+    private void processProject(final UUID projectUuid) throws Exception {
+        LOGGER.info("Submitting components of project %s for repository meta analysis".formatted(projectUuid));
+
+        try (final var qm = new QueryManager()) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java:74]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+                final long lastId = components.get(components.size() - 1).getId();
+                components = fetchNextComponentsPage(qm.getPersistenceManager(), null, lastId);
+            }
+        }
+
+        LOGGER.info("All components in portfolio submitted for repository meta analysis");
+    }
+
+    private List<Component> fetchNextComponentsPage(final PersistenceManager pm, final Project project, final Long lastId) throws Exception {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java:119]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/event/kafka/processor/RepositoryMetaResultProcessor.java [hunk 5]
```diff
@@ -1,40 +1,108 @@
...
+                            """.formatted(metaModel.getComponent()));
+                    return;
+                }
+
+                persistentRepoMetaComponent.setRepositoryType(RepositoryType.resolve(metaModel.getComponent().getPurl()));
+                persistentRepoMetaComponent.setNamespace(metaModel.getComponent().getPurl().getNamespace());
+                persistentRepoMetaComponent.setName(metaModel.getComponent().getPurl().getName());
+                persistentRepoMetaComponent.setLatestVersion(metaModel.getLatestVersion());
+                persistentRepoMetaComponent.setPublished(metaModel.getPublishedTimestamp());
+                persistentRepoMetaComponent.setLastCheck(new Date(record.timestamp()));
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/event/kafka/processor/RepositoryMetaResultProcessor.java:84]: @sonatype-lift ignore

**Review by VithikaS — APPROVED** [review]

**Pull request merged**